packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -71,18 +72,37 @@ function SelectionTool(props: Props) {
       }
 
       const { worldPt, sourceEvent } = evt;
-      const newEndPoint = worldToData(boundPointToFOV(worldPt, camera));
-      setEndPoint(newEndPoint);
+      const dataEndPoint = worldToData(boundPointToFOV(worldPt, camera));
+      setEndPoint(dataEndPoint);
+
+      const worldEndPoint = dataToWorld(dataEndPoint);
+      const htmlEndPoint = cameraToHtml(
+        new Vector3(worldEndPoint.x, worldEndPoint.y, 0).project(camera)
+      );
+
+      const worldStartPoint = dataToWorld(startPoint);
+      const htmlStartPoint = cameraToHtml(
+        new Vector3(worldStartPoint.x, worldStartPoint.y, 0).project(camera)
+      );
```

Comment:
On one hand I was thinking that you could have a function `processSelectionEvent` to compute all this like in `useCanvasEvents`, to avoid duplication. On the other hand, I noticed that the `worldStartPoint` and `htmlStartPoint` are recomputed unnecessarily, since they are already available in the initial `pointerDown` event. But then storing the initial `pointerDown` event in the state doesn't quite feel right. If you have any idea... Otherwise it's no big deal. :+1:

Reply:
We could compute `worldStartPoint` and `htmlStartPoint` with `useMemo`s from the state value `startPoint` outside of the handler ? :shrug:

Reply:
Yeah or even without `useMemo`, just above the event handlers so you can re-use the same variables.